application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/TeXCommand.java

```diff
@@ -56,42 +62,102 @@ public void onSlashCommand(@NotNull final SlashCommandEvent event) {
                     latex, e);
-            return;
+
+        } catch (IllegalStateException e) {
+            event.getHook().editOriginal(RENDERING_ERROR).queue();
+
+            logger.warn(
+                    "Unable to render latex, image does not have an accessible width or height. Formula was {}",
+                    latex, e);
+        }
+    }
+
+    private @NotNull Image renderImage(@NotNull TeXFormula formula) {
+        Image image = formula.createBufferedImage(TeXConstants.STYLE_DISPLAY, DEFAULT_IMAGE_SIZE,
+                FOREGROUND_COLOR, BACKGROUND_COLOR);
+
+        if (image.getWidth(null) == -1 || image.getHeight(null) == -1) {
+            throw new IllegalStateException("Image has no height or width");
         }
+        return image;
+    }
+
+    private void sendImage(@NotNull SlashCommandEvent event, @NotNull String userID,
+            @NotNull Image image) throws IOException {
+
+        ByteArrayOutputStream renderedTextImageStream = getRenderedTextImageStream(image);
         event.getHook()
             .editOriginal(renderedTextImageStream.toByteArray(), "tex.png")
             .setActionRow(Button.of(ButtonStyle.DANGER, generateComponentId(userID), "Delete"))
             .queue();
     }
 
+    @NotNull
+    private ByteArrayOutputStream getRenderedTextImageStream(@NotNull Image image)
+            throws IOException {
+
+        BufferedImage renderedTextImage = new BufferedImage(image.getWidth(null),
+                image.getHeight(null), BufferedImage.TYPE_4BYTE_ABGR);
+
+        renderedTextImage.getGraphics().drawImage(image, 0, 0, null);
+        ByteArrayOutputStream renderedTextImageStream = new ByteArrayOutputStream();
+
+        ImageIO.write(renderedTextImage, "png", renderedTextImageStream);
+
+        return renderedTextImageStream;
+    }
+
+    /**
+     * Converts inline latex like: {@code hello $\frac{x}{2}$ world} to full latex
+     * {@code \text{hello}\frac{x}{2}\text{ world}}.
+     */
+    @NotNull
+    private String convertInlineLatexToFull(@NotNull String latex) {
+        if (isInvalidInlineFormat(latex)) {
+            throw new ParseException(
+                    "The amount of $-symbols must be divisible by two. Did you forget to close an expression? ");
+        }
+
+        Matcher matcher = INLINE_LATEX_REPLACEMENT.matcher(latex);
+        StringBuilder sb = new StringBuilder();
+
+        while (matcher.find()) {
+            boolean isInsideMathRegion = matcher.group(1) != null;
+            if (isInsideMathRegion) {
+                sb.append(matcher.group(1).replace("$", ""));
+            } else {
+                sb.append("\\text{").append(matcher.group(2)).append("}");
+            }
+        }
+
+        return sb.toString();
+    }
+
+    private boolean isInvalidInlineFormat(String latex) {
+        return latex.chars().filter(ch -> ch == '$').count() % 2 == 1;
```

Comment:
Same here.
We should escape the possible escaped `\$` characters

Reply:
We discussed this and agreed that we can ignore it for now.

So unless you have a cheap and easy suggestion to adjust the code to support it, we can also go without it for now.

Reply:
Sorry about that.
Didn't know it was discussed.

But, wouldn't `(?<!\\)\$` be a good regex for it? It would only match unescaped dollar signs

Reply:
> Didn't know it was discussed.

No worries, not your fault. It was probably in Discord and not here.

Reply:
I believe it does.
It's a negative look behind regex.

We could also add a really simple unit test to test it too.
Not all tests need to be integration tests 😆

Reply:
![image](https://user-images.githubusercontent.com/13050270/155972757-45b18379-389a-4e46-9761-097bb2b10858.png)

Reply:
> We could also add a really simple unit test to test it too.

Yes, totally correct. We still have a separate GH issue open for adding tests to `/tex`, so I didnt want to bloat up this PR with more tasks than necessary (poor @IslamSakrak )

Reply:
I actually think that, as soon as this is merged, someone should tackle the unit tests for `/tex`. It would benefit a lot from it.

Reply:
You're correct.
I'm gonna mention this point on that issue, so it doesn't fall into the forgotten realm when those tests get addressed

Reply:
I wouldn't mind testing this also, I have written some tests already. Also, first time writing tests so might help me get better at it

Reply:
If you want, feel free to either create a second PR or also do it directly in this.